labs/SettingsControls/src/SettingsExpander/SettingsExpander.ItemsControl.cs

```diff
@@ -0,0 +1,72 @@
+
+namespace CommunityToolkit.Labs.WinUI;
+
+public partial class SettingsExpander
+{
+    public IList<object> Items
+    {
+        get { return (IList<object>)GetValue(ItemsProperty); }
+        set { SetValue(ItemsProperty, value); }
+    }
+
+    // Using a DependencyProperty as the backing store for Items.  This enables animation, styling, binding, etc...
+    public static readonly DependencyProperty ItemsProperty =
+        DependencyProperty.Register(nameof(Items), typeof(IList<object>), typeof(SettingsExpander), new PropertyMetadata(null, OnItemsConnectedPropertyChanged));
+
+    public object ItemsSource
+    {
+        get { return (object)GetValue(ItemsSourceProperty); }
+        set { SetValue(ItemsSourceProperty, value); }
+    }
+
+    public static readonly DependencyProperty ItemsSourceProperty =
+        DependencyProperty.Register(nameof(ItemsSource), typeof(object), typeof(SettingsExpander), new PropertyMetadata(null, OnItemsConnectedPropertyChanged));
+
+    public DataTemplate ItemTemplate
+    {
+        get { return (DataTemplate)GetValue(ItemTemplateProperty); }
+        set { SetValue(ItemTemplateProperty, value); }
+    }
+
+    // Using a DependencyProperty as the backing store for ItemTemplate.  This enables animation, styling, binding, etc...
+    public static readonly DependencyProperty ItemTemplateProperty =
+        DependencyProperty.Register(nameof(ItemTemplate), typeof(DataTemplate), typeof(SettingsExpander), new PropertyMetadata(null));
+
+    public StyleSelector ItemContainerStyleSelector
+    {
+        get { return (StyleSelector)GetValue(ItemContainerStyleSelectorProperty); }
+        set { SetValue(ItemContainerStyleSelectorProperty, value); }
+    }
+
+    public static readonly DependencyProperty ItemContainerStyleSelectorProperty =
+        DependencyProperty.Register(nameof(ItemContainerStyleSelector), typeof(StyleSelector), typeof(SettingsExpander), new PropertyMetadata(null));
+
+    private static void OnItemsConnectedPropertyChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs args)
+    {
+        if (dependencyObject is SettingsExpander expander && expander._itemsRepeater is not null)
+        {
+            var datasource = expander.ItemsSource;
+
+            if (datasource is null)
+            {
+                datasource = expander.Items;
+            }
+
+            expander._itemsRepeater.ItemsSource = datasource;
+        }
+    }
+
+    private void ItemsRepeater_ElementPrepared(MUXC.ItemsRepeater sender, MUXC.ItemsRepeaterElementPreparedEventArgs args)
+    {
+        if (ItemContainerStyleSelector != null &&
+            args.Element is FrameworkElement element &&
+            element.ReadLocalValue(FrameworkElement.StyleProperty) == DependencyProperty.UnsetValue)
+        {
+            // TODO: Get item from args.Index?
+            element.Style = ItemContainerStyleSelector.SelectStyle(null, element);
```

Comment:
Wonder if DataContext is set already here, should check, as that'd be just easier?